test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1686,6 +1682,9 @@ public void indexRandom(boolean forceRefresh, boolean dummyDocuments, boolean ma
                 client().admin().indices().prepareRefresh(indicesArray).setIndicesOptions(IndicesOptions.lenientExpandOpen()).get()
             );
         }
+        if (dummyDocuments) {
+            indexRandomForMultipleSlices(indicesArray);
+        }
```

Comment:
I wonder if there should be a different parameter that controls this? It seems like we are making some distinction with dummy docs for concurrent search and dummy docs not for concurrent search as we are ingesting dummy docs in 2 different ways (to be honest I don't understand why we have that distinction currently), so perhaps we should use a different parameter to control this here?

Reply:
yes i agree, however having a separate param would require changes in the usages of the current indexRandom function. 
Also, ideally ingesting docs for concurrent search tests should ideally be done by indexRandomForMultipleSlices and indexRandom function call should be replaced by indexRandomForMultipleSlices so then there is no need to have this if block in the indexRandom function. However, again there are a lot of places where the change has to be made.

Reply:
That makes sense to me -- should we use `indexRandomForConcurrentSearch` instead of `indexRandomForMultipleSlices` here though?

Reply:
Actually I thought on this some more -- `indexRandomForConcurrentSearch` doesn't work very well when this indexing happens before the setting is parameterized anyways so I guess it doesn't really make a difference.

Reply:
yes